File: action-type/13-template/go/yaml_config_loader.go
package main

import (
	"fmt"
)

type YamlConfigLoader struct {
	BaseConfigLoader
}

func (y *YamlConfigLoader) getReader(filePath string) interface{} {
	fmt.Println("Loading YAML config from:", filePath)
	return nil
}

func (y *YamlConfigLoader) parseConfig(reader interface{}) interface{} {
	fmt.Println("Parsing YAML config...")
	return struct{}{}
}

func (y *YamlConfigLoader) validateConfig(config interface{}) bool {
	fmt.Println("Validating YAML config...")
	return true
}

func (y *YamlConfigLoader) Load(filePath string) interface{} {
	return y.BaseConfigLoader.Load(filePath, y.getReader, y.parseConfig, y.validateConfig)
}

File: action-type/13-template/java/YamlConfigLoader.java

package main;

import java.io.Reader;

public class YamlConfigLoader extends AbstractConfigLoader<Object> {

    @Override
    protected Reader getReader(String filePath) {
        System.out.println("Loading YAML config from: " + filePath);
        // 实际实现应返回文件读取器
        return null;
    }

    @Override
    protected Object parseConfig(Reader reader) {
        System.out.println("Parsing YAML config...");
        // 实际实现应解析 YAML 并返回配置对象
        return new Object();
    }
}
